cpp/src/plasma/external_store.cc

```diff
@@ -0,0 +1,64 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <iostream>
+#include <sstream>
+
+#include "external_store.h"
+
+namespace plasma {
+
+std::string ExternalStores::ExtractStoreName(const std::string& endpoint) {
+  size_t off = endpoint.find_first_of(':');
+  if (off == std::string::npos) {
+    throw std::invalid_argument("Malformed endpoint " + endpoint);
```

Comment:
We use arrow::Status to indicate failures, in the future there might be an extensible API for this (see https://github.com/apache/arrow/pull/3199/files), but for now I'd suggest Status::Invalid here.